apps/dashboard/pages/api/v1/github/verify.ts

```diff
@@ -0,0 +1,211 @@
+import { Buffer } from "node:buffer";
+import crypto from "node:crypto";
+import type { Readable } from "node:stream";
+import { db } from "@/lib/db";
+import { env } from "@/lib/env";
+import { clerkClient } from "@clerk/nextjs";
+import { sha256 } from "@unkey/hash";
+import { Resend } from "@unkey/resend";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+export const config = {
+  api: {
+    bodyParser: false,
+  },
+  runtime: "nodejs",
+};
+
+type Key = {
+  key_identifier: string;
+  key: string;
+  is_current: boolean;
+};
+
+// Needs to be tested when Github is live
+const verifyGitSignature = async (
+  payload: string,
+  signature: string,
+  keyId: string,
+  GITHUB_KEYS_URI: string,
+) => {
+  const gitHub = await fetch(GITHUB_KEYS_URI);
+  if (!gitHub.ok) {
+    console.error("Github verify error");
+    return false;
+  }
+  const gitBody = await gitHub.json();
+  const publicKey =
+    gitBody.public_keys.find((k: Key) => {
+      if (k.key_identifier === keyId) {
+        return k;
+      }
+    }) ?? null;
+
+  if (!publicKey) {
+    console.error("No public key found");
+  }
+  const verify = crypto.createVerify("SHA256").update(payload);
+  return !verify.verify(publicKey.key, Buffer.from(signature.toString(), "base64"));
+};
+
+export default async function handler(request: NextApiRequest, response: NextApiResponse) {
+  const { RESEND_API_KEY, GITHUB_KEYS_URI } = env();
+
+  if (!RESEND_API_KEY || !GITHUB_KEYS_URI) {
+    console.error("Missing required environment variables");
+    return response.status(201).json({});
```

Comment:
201 means success, we should return a 500

Reply:
changed to response.status(500).json({error: "Internal Server Error" });